core/chains/evm/txmgr/eth_broadcaster.go

```diff
@@ -654,7 +641,9 @@ func (eb *EthBroadcaster) handleInProgressEthTx(ctx context.Context, etx EthTx,
 		// Despite the error, the RPC node considers the previously sent
 		// transaction to have been accepted. In this case, the right thing to
 		// do is assume success and hand off to EthConfirmer
-		return eb.saveAttempt(&etx, attempt, EthTxAttemptBroadcast), true
+		return eb.orm.UpdateEthTxAttemptInProgressToBroadcast(&etx, attempt, EthTxAttemptBroadcast, func(tx pg.Queryer) error {
+			return eb.incrementNonceAtomic(tx, etx)
```

Comment:
The next nonce increment must happen in the same atomic transaction as updating eth tx attempt to broadcast. So we have to pass in the transaction to the keystore orm method (inside eb.incrementNonceAtomic)